plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,490 @@
+description:
+- Manage IP Service Level Agreement (SLA) Track Lists on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Samita Bhattacharjee (@samitab)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a tenant template.
+    type: str
+    required: true
+  ipsla_track_list:
+    description:
+    - The name of the IPSLA Track List.
+    type: str
+    aliases: [ name ]
+  description:
+    description:
+    - The description of the IPSLA Track List.
+    type: str
+  ipsla_track_list_uuid:
+    description:
+    - The UUID of the IPSLA Track List.
+    - This parameter is required when the O(ipsla_track_list) attribute needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  type:
+    description:
+    - The IPSLA Track List type used for determining up or down status.
+    - This parameter is required when creating the IPSLA Track List.
+    type: str
+    choices: [ percentage, weight ]
+  threshold_up:
+    description:
+    - The IPSLA Track List percentage or weight up threshold.
+    - The value must be in the range 0 - 100 when O(type=percentage).
+    - The value must be in the range 0 - 255 when O(type=weight).
+    - The value must be greater than or equal to O(threshold_down).
+    - The default value is 1.
+    type: int
+    aliases: [ up ]
+  threshold_down:
+    description:
+    - The IPSLA Track List percentage or weight down threshold.
+    - The value must be in the range 0 - 100 when O(type=percentage).
+    - The value must be in the range 0 - 255 when O(type=weight).
+    - The value must be less than or equal to O(threshold_up).
+    - The default value is 0.
+    type: int
+    aliases: [ down ]
+  members:
+    description:
+    - The IPSLA Track List members.
+    - Providing a new list of O(members) will replace the existing members from the IPSLA Track List.
+    - Providing an empty list will remove the O(members=[]) from the IPSLA Track List.
+    type: list
+    elements: dict
+    suboptions:
+      destination_ip:
+        description:
+        - The destination IP of the member.
+        - Must be a valid IPv4 or IPv6 address.
+        type: str
+        required: true
+        aliases: [ ip ]
+      weight:
+        description:
+        - The weight of the member.
+        - The default value is 10.
+        type: int
+      ipsla_monitoring_policy_uuid:
+        description:
+        - The UUID of the IPSLA Monitoring Policy to use for the member.
+        - This parameter can be used instead of O(members.ipsla_monitoring_policy).
+        type: str
+      ipsla_monitoring_policy:
+        description:
+        - The name IPSLA Monitoring Policy to use for the member.
+        - This parameter can be used instead of O(members.ipsla_monitoring_policy_uuid).
+        type: str
+        aliases: [ ipsla_monitoring_policy_name ]
```

Comment:
If we decide to input uuid for these nested inputs, should we then do it for all objects that have an uuid? think template in that case also has an uuid.

Reply:
Do you mean template within the scope sub-option or the top level template?

The scope template is only required if the L3out or BD UUID isn't provided. I think having scope template UUID as an options might make the module a little more complex than it needs to be.

If template UUID is an option at the top level then we would have to make it consistent everywhere and split that into a separate task.

Reply:
Any of them that allow uuid input I mean. The original uuid input was only to change a name of the object. Later a decision was made to do it for more, but now I see we then do not do it for everything, thus my question.

Regarding complexity, I think the uuid as input anywhere besides the name change use-case already makes things more complex than it needs to be.

Reply:
The main reason to provide UUID as an option for any input, apart from the name change case, would be if there is a performance benefit. In this module using UUID instead of template/schema/name for `scope` is much quicker. This is because we would use the UUID directly instead of having to search for UUID from a potentially different template.

I think UUID for IPSLA Monitoring Policy would been useful if it could reference a policy from a different template. Initially I thought this was the case but its not and now I can remove `ipsla_monitoring_policy_uuid`.

Reply:
Removed `ipsla_monitoring_policy_uuid`.